crates/server/src/store/in_memory/authority.rs

```diff
@@ -427,7 +428,7 @@ impl InnerInMemory {
             // we need to change the name to the query name in the result set since this was a wildcard
             .map(|rrset| {
                 let mut new_answer =
-                    RecordSet::new(name.borrow(), rrset.record_type(), rrset.ttl());
+                    RecordSet::with_ttl(Name::from(name), rrset.record_type(), rrset.ttl());
```

Comment:
Ah, this makes more sense... I almost want to make types for each of these to avoid this type of mistake in the future. Thanks for fixing this.

Reply:
No problem ! Yeah some newtype can be handy but they are also a chore it's a balance I suppose.